File: Easy/Python/121. Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        buy = 0
        sell = 1
        maximum = 0

        # Use a sliding window to find the maximum
        while sell < len(prices):
            if prices[buy] < prices[sell]:
                profit = prices[sell] - prices[buy]
                maximum = max(maximum, profit)
            else:
                buy = sell
            sell += 1

        return maximum

File: Easy/Java/121. Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = 0;
        int sell = 1;
        int maximum = 0;

        // Use a sliding window to find the maximum
        while(sell < prices.length){
            if(prices[buy] < prices[sell]){
                int profit = prices[sell] - prices[buy];
                maximum = Math.max(maximum, profit);
            }else{
                buy = sell;
            }
            sell++;
        }

        return maximum;
    }
}
